schemas/sublime-project.json

```diff
@@ -1,7 +1,88 @@
 {
+  "$schema": "http://json-schema.org/draft-07/schema#",
   "$id": "sublime://sublime-project",
   "title": "Sublime Text Project",
   "allowComments": true,
   "allowTrailingCommas": true,
-  "type": "object"
+  "type": "object",
+  "properties": {
+    "additionalProperties": false,
```

Comment:
I don't agree with this decision to disallow additional properties. Various plugins out in the wild may use a sublime-project to store arbitrary data. Same goes for sublime-build files.

Disallowing additional properties in sublime-keymap, sublime-mousemap and sublime-completion files make sense though.

Reply:
But plugins add those to `settings` object, don't they? And `settings` object allows additional properties.

Reply:
As for `sublime-build`, do you have any examples where this is done currently?

Reply:
For sublime-build files, the value for the `"target"`key is a `WindowCommand`. It receives all key-value pairs of the invoked build target in `**kwargs`.
For instance, I utilize this fact in my CMakeBuilder plugin to pass along build artefact information that I store in the ST build system.

For sublime-project files, you can get the entire content via `window.project_data()`. It's a Python dictionary. You can put anything in there. And then you can store it with `window.set_project_data(...)`. ST3 won't complain if there are unknown keys.

Reply:
> For sublime-build files, the value for the `"target"`key is a `WindowCommand`.

From the schema's perspective, it's a string through, no?
And `target` is already in the schema: https://github.com/sublimelsp/LSP-json/blob/2f72c1ece781b30696c3c391a84d6a7854b7317e/schemas/sublime-build.json#L47-L50

> For sublime-project files, you can get the entire content via `window.project_data()`. It's a Python dictionary.

Actually for `sublime-project.json`, the `additionalProperties` key is in the wrong place which makes it work as expected. It's defined inside `properties` object which basically means that it defines a property called literally `additionalProperties` rather than functioning as limiter. If it would be defined in the root of the schema then it would actually restrict properties that are allowed on the object.
So, it works as expected right now. I'll just remove that key.

Reply:
> From the schema's perspective, it's a string though, no?

Ah, yes. I did not explain myself very well. You can have a build system like this:

```js
{
  "target": "make_thing",
  "foo": "bar",
  "selector": "source.python",
  "some_number": 42
}
```

and define a command like this:

```python
import sublime_plugin


class MakeThingCommand(sublime_plugin.WindowCommand):

    def run(self, *args, **kwargs):
        foo = kwargs.get("foo")  # will have the value "bar"
        selector = kwargs.get("selector")  # will have the value "source.python"
        some_number = kwargs.get("some_number")  # will have the value 42
```

Reply:
So basically, ST does this: If `"target"` is present, then invoke the corresponding command with the rest of the key-value pairs. Otherwise, invoke `ExecCommand` with the rest of the key-value pairs.

Reply:
OK, I get it now. But in fact, sublime-build schema doesn't and never had `additionalProperties: false` so I think we are good. :)